wordpress-mobile/WordPress-iOS
# Notifications: add nav bar button to mark all as read

Ref: #17135 
WPKit PR: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/476

In the Notifications view, a checkmark button is added to the nav bar to mark all unread notifications as read for the selected filter.

This is to merge Alp's trial project PRs to `trunk`.
- #17772
- #17789
- #17796
- #17808

To test:
- Go to Notifications.
- Verify a checkmark icon now appears in the nav bar to the left of the settings button.
- For filters that do not have unread notifications, the checkmark is disabled.
- For filters that do have unread notifications, the checkmark is enabled.
- Tip: if you don't have any unread notifications, swipe right on a notification row to mark it unread.

| <img width="446" alt="read" src="https://user-images.githubusercontent.com/1816888/150582898-29123d0c-cca4-424a-a025-e83ab899a23f.png"> | <img width="448" alt="unread" src="https://user-images.githubusercontent.com/1816888/150582789-1344252a-c03d-49a5-aafd-efd2c0ab70fd.png"> |
|--------|-------|


- On a filter that has unread notifications, tap the checkmark button.
- Verify all the notifications in that filter are marked as read, and the button is disabled.
- Verify notifications in other filters are not.
- Note that, since the button applies to the selected filter, `All` and `Unread` will mark all unread notifications as read (thus affecting the other filters).

## Regression Notes
1. Potential unintended areas of impact
It's possible marking a single notification as read/unread could be affected.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Verified marking a single notification as read/unread still works as expected. And there are existing tests that passed.

3. What automated tests I added (or what prevented me from doing so)
Tests were added to WPKit.

PR submission checklist:
